geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java

```diff
@@ -253,7 +253,7 @@ private void createIndexOnExistingRegion(PartitionedRegion region, String indexN
 
     try {
       validateLuceneIndexProfile(region);
-    } catch (Exception e) {
+    } catch (IllegalStateException e) {
```

Comment:
I think you should probably include info on which tests cover this code and consider adding a test for any change including this one.